plugins/kapt3/kapt3-compiler/src/org/jetbrains/kotlin/kapt3/stubs/ClassFileToSourceStubConverter.kt

```diff
@@ -751,6 +752,11 @@ class ClassFileToSourceStubConverter(val kaptContext: KaptContextForStubGenerati
         val value = field.value
 
         val origin = kaptContext.origins[field]
+        val isConstantValue = when (val declaration = origin?.element) {
+            is KtProperty -> declaration.hasModifier(KtTokens.CONST_KEYWORD)
```

Comment:
This is effectively a breaking change, so I'm wondering if we should treat:
- `const`
- any kind of primitive + String values for a `val`
as constants and inline them?

This ensures existing users do not break.